src/utils/metrics/reporter.go

```diff
@@ -0,0 +1,120 @@
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"go.uber.org/zap"
+)
+
+// Reporter gathers metrics across jobs and reports them.
+// Concurrency-safe.
+type Reporter interface {
+	// WriteSummary dumps Reporter contents into the target.
+	WriteSummary()
+
+	SumBytesSent() uint64
```

Comment:
I think it's better to have generic Sum(stat) interface in case we have usage for other metrics outside the package

Reply:
Exposed the old full Sum() function